### PR TITLE
Make code and test compatible with new version of mflike and Cobaya 

### PR DESCRIPTION
**Pull request by paganol**
This PR addresses issue #20 and other incompatibilities:
- We updated mflike to the new version, which includes a bunch of new nuisance parameters
- We fixed the test for mflike 
- We fixed the compatibility of cobaya to version 3.1.1

### CODE REVIEW

File: soliket/mflike.py [hunk 1]
```diff
@@ -30,22 +28,22 @@ class MFLike(GaussianLikelihood, _InstallableLikelihood):
...
-                "'path' or the common property '%s'.",
-                _packages_path,
-            )
+        if ((not getattr(self, "path", None)) and
+                (not getattr(self, "packages_path", None))):
+            raise LoggedError(self.log,
+                              "No path given to MFLike data. "
+                              "Set the likelihood property "
+                              "'path' or the common property '%s'.",
+                              _packages_path)
```
Comment (itrharrison) [inline soliket/mflike.py:42]: `_packages_path` is no longer set, so this may fail. Is it defined elsewhere?

File: soliket/lensing/lensing.py [hunk 2]
```diff
@@ -3,16 +3,15 @@
 
 import numpy as np
 
-from cobaya.likelihoods._base_classes import _InstallableLikelihood
-from cobaya.conventions import _packages_path
```
Comment (itrharrison) [inline soliket/lensing/lensing.py:7]: `_packages_path` is no longer set, and appears on l48 so this may fail. Is it defined elsewhere?

**Review by itrharrison — CHANGES_REQUESTED** [review]
This looks good in both fixing the bugs with fgspectra installation, and with the `cobaya.conventions` in the new cobaya version, as well as doing some work with the new mflike.

Possibly in future it would be better to do the bug fixes as separate PRs which go in before the new functionality with mflike, but should not be a big problem for here as I don't think there is any conflicting work done elsewhere, apart from Jack's PR, but I think that can be superseded without too much wasted effort.

**Comment by paganol** [comment]
Hi @itrharrison the last push should solve the issue you mentioned

**Review by itrharrison — APPROVED** [review]

**Pull request merged**